### PR TITLE
API-1432 create event dispatcher observer test service

### PR DESCRIPTION
**Pull request by pchasle**
Create a new test service that wrap the EventDispatcher to keep in-memory dispatched events.
Useful to test that all StorageEvents are really dispatched in Integration and e2e tests.

### CODE REVIEW

File: tests/back/Integration/IntegrationTestsBundle/EventDispatcher/EventDispatcherObserver.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private function incrementStorageEvents(string $eventName, GenericEvent $event): void
+    {
+        if (!isset($this->storageEvents[$eventName][get_class($event->getSubject())])) {
+            $this->storageEvents[$eventName][get_class($event->getSubject())] = 0;
+        }
+
+        $this->storageEvents[$eventName][get_class($event->getSubject())] += 1;
+    }
+
+    public function getStorageEventCount(string $eventName, string $className): int
```
Comment (pchasle) [inline tests/back/Integration/IntegrationTestsBundle/EventDispatcher/EventDispatcherObserver.php:40]: We need to be able to determine if those save events are creation or update.

File: tests/legacy/features/pim/enrichment/product/import/import_products.feature [hunk 2]
```diff
@@ -65,6 +65,7 @@ Feature: Import products coming from an external application
       """
     When the products are imported via the job csv_footwear_product_import
     Then there should be 1 product
+    And the PIM should be notified of the creation of 1 products
```
Comment (pchasle) [inline tests/legacy/features/pim/enrichment/product/import/import_products.feature:68]: Doesn't work, we are probably just seeing the product from
```
Given the following product:
      | sku     | name-en_US |	      | sku     | name-en_US |
      | SKU-001 | FooBar     |	      | SKU-001 | FooBar     |
```

File: config/services/behat/services.yml [hunk 3]
```diff
@@ -55,3 +55,13 @@ services:
...
+
+    akeneo_integration_tests.event_dispatcher_observer:
+        class: 'Akeneo\Test\IntegrationTestsBundle\EventDispatcher\EventDispatcherObserver'
+        decorates: event_dispatcher
+        decoration_inner_name: event_dispatcher.inner
+        arguments:
+            - '@event_dispatcher.inner'
+            - '@debug.stopwatch'
+            - '@logger'
+            - '@request_stack'
```
Comment (tomglvng) [inline config/services/behat/services.yml:67]:
```suggestion
            - '@request_stack'
            
```

**Comment by tseho** [comment]
done https://github.com/akeneo/pim-community-dev/pull/13858

**Pull request closed without merging**